File: elvitrina-frontend/src/app/core/models/messages/friendrequest.ts
import { User } from "../user/user.model";

export class FriendRequest {
  id: number;
  sender: User;
  receiver: User;
  senderId: number;
  receiverId: number;
  senderFirstName: string; 
  senderLastName: string;  
  receiverFirstName: string; 
  receiverLastName: string;  
  status: 'PENDING' | 'ACCEPTED' | 'DECLINED';
  sentAt: string;
}

File: elvitrina-backend/src/main/java/com/sudoers/elvitrinabackend/model/entity/FriendRequest.java
package com.sudoers.elvitrinabackend.model.entity;

import com.sudoers.elvitrinabackend.model.enums.RequestStatus;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Entity
@Data
public class FriendRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "sender_id", nullable = false)
    private User sender;

    @ManyToOne
    @JoinColumn(name = "receiver_id", nullable = false)
    private User receiver;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RequestStatus status;

    private LocalDateTime sentAt;
}